internal/util/validation/validation.go

```diff
@@ -94,6 +94,47 @@ func ValidateString(s *string, path string, minLen int, maxLen int, patternRegex
 	return asErrors(errs)
 }
 
+func ValidateCronField(field string, min, max int, fieldName string) error {
```

Comment:
Can this be handled more efficiently using a compiled regex with capturing groups?

Reply:
Great question, under the hood these string evaluations are highly optimized. In general, unlike some languages like Perl regex is really not the best for performance even when compiled.  

For the test I came up with this monster which I believe doesn't even cover everything.

```
regex := regexp.MustCompile(`^
    (\*|[0-5]?\d|\d+-\d+|[0-5]?\d/\d+) # Minutes
    (\s+) 
    (\*|[0-5]?\d|\d+-\d+|[0-5]?\d/\d+) # Hours
    (\s+) 
    (\*|[1-2]?\d|3[0-1]|\d+-\d+|[1-2]?\d/\d+) # Day of month
    (\s+)
    (\*|1[0-2]|[1-9]|\d+-\d+|1[0-2]/\d+) # Months
    (\s+) 
    (\*|[0-6]|\d+-\d+|[0-6]/\d+)  # Day of week
$`)
```
Then created these benches with a simple test case

func BenchmarkValidateCronSchedule(b *testing.B) {
	require := require.New(b)
	cron := "* 1 * * *"
	for i := 0; i < b.N; i++ {
		err := ValidateCronSchedule(cron)
		require.NoError(err)
	}
}

func BenchmarkValidateCronScheduleWithRegex(b *testing.B) {
	require := require.New(b)
	cron := "* 1 * * *"
	for i := 0; i < b.N; i++ {
		err := ValidateCronScheduleWithRegex(cron)
		require.NoError(err)
	}
}
```

The output shows the regex was quite a bit slower.

```
cpu: Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
BenchmarkValidateCronSchedule-12    	  750240	      1771 ns/op	      96 B/op	       2 allocs/op
```

```
cpu: Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
BenchmarkValidateCronScheduleWithRegex-12    	   12692	    101958 ns/op	   39783 B/op	     210 allocs/op
```

Reply:
🚀 ! Any chance to add a comment as example of an input?

Reply:
Saw another issue this function is actually cruft from my first pass and will be removed in favor of `ValidateCronSchedule` which was used in the bench.

Reply:
I ended up just taking a dep on this to github.com/robfig/cron in the end